cartography/intel/trivy/scanner.py

```diff
@@ -361,3 +374,61 @@ def sync_single_image_from_s3(
             f"Failed to process S3 scan results for {image_uri} from {s3_object_key}: {e}"
         )
         raise
+
+
+@timeit
+def read_scan_results_from_file(
+    file_path: str,
+) -> tuple[str | None, list[dict], str | None]:
+    """Read and parse Trivy scan results from a local file."""
+    logger.debug(f"Reading scan results from file: {file_path}")
+    with open(file_path, encoding="utf-8") as f:
+        trivy_data = json.load(f)
+
+    artifact_name = trivy_data.get("ArtifactName")
```

Comment:
Ah ok, so this from trivy is basically an ECR image URI: https://github.com/cartography-cncf/cartography/blob/b68f2e372ba12a474a490a90eae33d08c87e883b/tests/data/trivy/trivy_sample.py#L4.

I think we should do a `trivy_data["ArtifactName"]` instead of a `.get()` and raise exc if this is None because `ArtifactName` feels like a super important field and if it's missing then that sounds like something has gone very wrong and we shouldn't continue.

It'd also simplify the code to make `artifact_name` not be an optional field anymore.

Reply:
done